graphql/src/execution/execution.rs

```diff
@@ -67,6 +74,151 @@ where
             deadline: self.deadline,
         }
     }
+
+    /// See https://developer.github.com/v4/guides/resource-limitations/.
+    ///
+    /// If the query is invalid, returns `Ok(0)` so that execution proceeds and
+    /// gives a proper error.
+    pub(crate) fn root_query_complexity(
+        &self,
+        root_type: &s::TypeDefinition,
+        root_selection_set: &q::SelectionSet,
+    ) -> Result<u64, QueryExecutionError> {
+        const MAX_DEPTH: u32 = 100;
+
+        match self.query_complexity(root_type, root_selection_set, MAX_DEPTH, 0) {
+            Ok(complexity) => Ok(complexity),
+            Err(ComplexityError::Invalid) => Ok(0),
+            Err(ComplexityError::TooDeep) => Err(QueryExecutionError::TooDeep),
+            Err(ComplexityError::Overflow) => {
+                Err(QueryExecutionError::TooComplex(u64::max_value(), 0))
```

Comment:
Is the `0` here the right thing to pass to `TooComplex`? Intuitively I'm thinking this should be the limit?

Reply:
These are abitrary values. A query has to be so huge to hit this case that I don't want to write any additional code to support it. This function doesn't have `max_complexity` as a parameter so that I can easily put it here.

Reply:
Oh, of course, this is the overflow case, which should rarely be hit. Makes sense.